Source/Thunder/Controller.cpp

```diff
@@ -1227,13 +1227,25 @@ namespace Plugin {
 
             std::list< IMetadata::Data::Proxy> elements;
 
+            int status = 0;
             for (const ProxyStub::UnknownProxy* proxy : collection) {
                 IMetadata::Data::Proxy data;
                 data.Instance = proxy->Implementation();
                 data.Interface = proxy->InterfaceId();
                 data.Count = proxy->ReferenceCount();
-                data.Name = proxy->Name();
 
+                #ifdef __LINUX__
```

Comment:
This sounds/looks like code duplication ;-) Time for a new method in Portability like string Demangle(const TCHAR name[]) , cause this code is also here: https://github.com/rdkcentral/Thunder/blob/master/Source/core/Portability.cpp#L227-L244, Tha do not call it here bt there where the proxy->Name() is set, s every one using the ProxyStub::UnknownProxy type can benefit form this work...